src/com/facebook/buck/jvm/java/AbstractJavacPluginProperties.java

```diff
@@ -39,6 +39,14 @@
 @Value.Immutable
 @BuckStyleImmutable
 abstract class AbstractJavacPluginProperties implements AddsToRuleKey {
+
+  enum Type {
```

Comment:
Do subclasses instead of an enum make less sense?

Reply:
I don't think so. There would have to have a big refactor in order to the rest of the ecosystem to understand this. As far as I saw, this was implemented as JavacPluginProperties (and not JavaAnnotationProcessorProperties) since the beginning waiting for when javac plugins were implemented as well.